### PR TITLE
Import Makefile.dapper downloaded from Shipyard

### PR DESCRIPTION
**Pull request by mkolesnik**
Instead of redefining the dapper file here, download it from Shiyard

Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>

### CODE REVIEW

File: Makefile.dapper [hunk 1]
```diff
@@ -1,19 +1,8 @@
...
+# This Makefile imports the rules to run in a Dapper-based build environment.
+# The rules are being imported via download from Shipyard where the base file can be found
 
-.dapper:
-	@echo Downloading dapper
-	@curl -sL https://releases.rancher.com/dapper/latest/dapper-`uname -s`-`uname -m` > .dapper.tmp
-	@@chmod +x .dapper.tmp
-	@./.dapper.tmp -v
-	@mv .dapper.tmp .dapper
+BASE_BRANCH := devel
```
Comment (skitt) [inline Makefile.dapper]:
```suggestion
BASE_BRANCH ?= devel
```

File: Makefile.dapper [hunk 2]
```diff
@@ -1,19 +1,8 @@
...
+_ := $(shell test -f Makefile.shipyard || curl -L https://raw.githubusercontent.com/submariner-io/shipyard/$(BASE_BRANCH)/Makefile.dapper > Makefile.shipyard)
```
Comment (skitt) [inline Makefile.dapper]: How do we ensure that local builds don’t use a stale file?

**Review by skitt — CHANGES_REQUESTED** [review]

File: Makefile.dapper (hunk 2 above)
Comment (mkolesnik) [reply Makefile.dapper]: Well the intention was to minimize local redownload each time, however I see were also fetching dapper each time.. So.. Maybe it's unnecessary to optimize right now?

Comment (mkolesnik) [reply Makefile.dapper]: But generally what I thought of is to have `make clean` which also removes these cached files, theoretically :)

Comment (mkolesnik) [reply Makefile.dapper]: > Well the intention was to minimize local redownload each time, however I see were also fetching dapper each time.. So.. Maybe it's unnecessary to optimize right now?

Actually just realized I missed the point of `make` itself caching it locally, so its the same there I guess..

File: Makefile.dapper (hunk 1 above)
Comment (mkolesnik) [reply Makefile.dapper]: Why would we want to allow it to be set from outside?

**Comment by mkolesnik** [comment]
@skitt OK this is a much more "Makefile-ish" approach, let me know what you think

File: Makefile [hunk 3]
```diff
@@ -84,9 +86,15 @@ else
 
 # Not running in Dapper
 
+Makefile.dapper:
+	@echo Downloading $@
+	@curl -sfL -o $@ https://raw.githubusercontent.com/submariner-io/shipyard/$(BASE_BRANCH)/$@
```
Comment (skitt) [inline Makefile]:
```suggestion
	@curl -sfL -O https://raw.githubusercontent.com/submariner-io/shipyard/$(BASE_BRANCH)/$@
```

**Review by skitt — CHANGES_REQUESTED** [review]
Much better, thanks! A minor tweak if I may...

File: Makefile.dapper (hunk 2 above)
Comment (skitt) [reply Makefile.dapper]: Right, `make clean` would be nice to have! (In another PR, not necessarily by you.)

File: Makefile (hunk 3 above)
Comment (mkolesnik) [reply Makefile]: Thanks!
Had to do it manually to avoid "Commit suggestion" commit since this PR has 2 distinct commits (though very simple ones indeed)

**Review by skitt — APPROVED** [review]

**Review by dfarrell07 — APPROVED** [review]

**Review by mangelajo — APPROVED** [review]

**Pull request merged**